pkg/envoy/ads/response_benchmark_test.go

```diff
@@ -0,0 +1,211 @@
+	cert, _ := certificate.DecodePEMCertificate(certPEM.GetCertificateChain())
+	server, _ = tests.NewFakeXDSServer(cert, nil, nil)
+
+	proxyUUID := uuid.New()
+	labels := map[string]string{constants.EnvoyUniqueIDLabelName: proxyUUID.String()}
+	meshSpec := smi.NewSMIClient(informerCollection, tests.OsmNamespace, kubeController, msgBroker)
+	mc := catalog.NewMeshCatalog(
+		kubeController,
+		meshSpec,
+		certManager,
+		policyController,
+		stop,
+		osmConfigurator,
+		[]service.Provider{kubeProvider},
+		[]endpoint.Provider{kubeProvider},
+		msgBroker,
+	)
+
+	proxyRegistry := registry.NewProxyRegistry(registry.ExplicitProxyServiceMapper(func(*envoy.Proxy) ([]service.MeshService, error) {
+		return nil, nil
+	}), nil)
+
+	pod := tests.NewPodFixture(namespace, fmt.Sprintf("pod-0-%s", proxyUUID), tests.BookstoreServiceAccountName, tests.PodLabels)
+	pod.Labels[constants.EnvoyUniqueIDLabelName] = proxyUUID.String()
+	_, err = kubeClient.CoreV1().Pods(namespace).Create(context.Background(), pod, metav1.CreateOptions{})
+	if err != nil {
+		b.Fatalf("Failed to create pod: %v", err)
+	}
+
+	// monitor namespace
+	nsObj := corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   namespace,
+			Labels: map[string]string{constants.OSMKubeResourceMonitorAnnotation: tests.MeshName},
+		},
+	}
+	if err := informerCollection.Add(informers.InformerKeyNamespace, &nsObj, &testing.T{}); err != nil {
+		b.Fatalf("Failed to add namespace to informer collection: %s", err)
+	}
+
+	svc := tests.NewServiceFixture(proxyService.Name, namespace, labels)
+	_, err = kubeClient.CoreV1().Services(namespace).Create(context.Background(), svc, metav1.CreateOptions{})
+	if err != nil {
+		b.Fatalf("Failed to create service: %v", err)
+	}
+
+	proxy = envoy.NewProxy(envoy.KindSidecar, proxyUUID, proxySvcAccount.ToServiceIdentity(), nil)
+
+	adsServer = NewADSServer(mc, proxyRegistry, true, tests.Namespace, osmConfigurator, certManager, kubeController, nil)
+}
+
+func BenchmarkSendXDSResponse(b *testing.B) {
+	testCases := []struct {
+		name    string
+		xdsType envoy.TypeURI
+	}{
+		{
+			name:    "ADS",
+			xdsType: envoy.TypeADS,
+		},
+		{
+			name:    "LDS",
+			xdsType: envoy.TypeLDS,
+		},
+		{
+			name:    "SDS",
+			xdsType: envoy.TypeSDS,
+		},
+		{
+			name:    "RDS",
+			xdsType: envoy.TypeRDS,
+		},
+		{
+			name:    "CDS",
+			xdsType: envoy.TypeCDS,
+		},
+	}
+
+	if err := logger.SetLogLevel("error"); err != nil {
+		b.Logf("Failed to set log level to error: %s", err)
+	}
+
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			setupTestServer(b)
+
+			// Set subscribed resources for SDS
+			proxy.SetSubscribedResources(tc.xdsType, mapset.NewSetWith("service-cert:default/bookstore", "root-cert-for-mtls-inbound:default/bookstore"))
+
+			b.ResetTimer()
+			b.StartTimer()
+			for i := 0; i < b.N; i++ {
+				if err := adsServer.sendResponse(proxy, &server, nil, osmConfigurator, envoy.XDSResponseOrder...); err != nil {
```

Comment:
instead of passing in envoy.XDSResponseOrder, you should pass in tc.xdsType for more clarity